jib-core/src/main/java/com/google/cloud/tools/jib/configuration/PortsWithProtocol.java

```diff
@@ -79,6 +79,30 @@ public Protocol getProtocol() {
     return protocol;
   }
 
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
+    if (other == null || other.getClass() != PortsWithProtocol.class) {
+      return false;
+    }
+    PortsWithProtocol otherPort = (PortsWithProtocol) other;
+    return minPort == otherPort.minPort
+        && maxPort == otherPort.maxPort
+        && protocol == otherPort.protocol;
+  }
+
+  @Override
+  public int hashCode() {
+    return (protocol == Protocol.TCP ? 1 : -1) * minPort + (maxPort << 16);
```

Comment:
"0-1024/udp" and "0-1024/tcp" ends up with the same code. How about `Objects.hash(minPort, maxPort, protocol)`? Not that I know now `Objects.hash()` works.

Reply:
Hmm yeah this seems like a better idea

Reply:
Just to be clear. I wrongly assumed for a moment that `hashCode()` should be unique for each combination of the fields. So, technically, your original implementation would probably work OK, even with the possible overflow and such.

Reply:
+1